lib/pbench/server/api/auth.py

```diff
@@ -0,0 +1,129 @@
+
+    def get_auth_token(self, logger):
+        # get auth token
+        auth_header = request.headers.get("Authorization")
+
+        if not auth_header:
+            logger.warning("Missing expected Authorization header")
+            abort(
+                403,
+                message="Please add 'Authorization' token as Authorization: Bearer <JWT_Auth_token>",
+            )
+
+        try:
+            auth_schema, auth_token = auth_header.split()
+        except ValueError:
+            logger.warning("Malformed Auth header during logout")
+            abort(
+                401,
+                message="Malformed Authorization header, please add request header as Authorization: Bearer <session_token>",
+            )
+        else:
+            if auth_schema.lower() != "bearer":
+                logger.warning(
+                    "Expected authorization schema to be 'bearer', not '{}'",
+                    auth_schema,
+                )
+                abort(
+                    401,
+                    message="Malformed Authorization header, please add request header as Authorization: Bearer <session_token>",
+                )
+            return auth_token
+
+    def verify_auth(self):
+        @Auth.token_auth.verify_token
+        def verify_token(auth_token):
+            """
+            Validates the auth token
+            :param auth_token:
+            :param app:
+            :return: integer|string
+            """
+            try:
+                payload = jwt.decode(
+                    auth_token,
+                    os.getenv("SECRET_KEY", "my_precious"),
+                    algorithms="HS256",
+                )
+                user_id = payload["sub"]
+                if ActiveTokens.valid(auth_token):
+                    user = User.query(id=user_id)
+                    return user
+                return False
+            except jwt.ExpiredSignatureError:
+                try:
+                    ActiveTokens.delete(auth_token)
+                except Exception:
+                    Auth.logger.error(
+                        "User attempted Pbench expired token but we could not delete the expired auth token from the database. token: '{}'",
+                        auth_token,
+                    )
+                    return False
+                Auth.logger.warning(
+                    "User attempted Pbench expired token '{}', Token deleted from the database and no longer tracked",
+                    auth_token,
+                )
+                return False
+            except jwt.InvalidTokenError:
+                Auth.logger.warning(
+                    "User attempted invalid Pbench token '{}'", auth_token
+                )
+                return False
+            except Exception:
+                Auth.logger.exception(
+                    "Exception occurred while verifying the auth token '{}'", auth_token
+                )
+                return False
```

Comment:
You didn't actually make this change (instead, you modified each individual `return` statement)...was that intentional?

Reply:
How would I have the common return from multiple exception, the pointer will never go to the end of the function if the exception occurred

Reply:
If you don't take any explicit action (such as `raise`, `return`, `continue`, `break`, etc.), then control will transfer to the first statement after the last `except` clause (or after the attached `else` clause, if it is present).  That is, `try`/`except` works a lot like `if`/`else` -- control just falls out the bottom.  (The magic happens when none of the `except` clauses match (or when there is a `finally` clause...); otherwise, the exception is considered to have been "handled", and execution resumes normally in the outer block.)

```
>>> def foo():
...   try:
...     print("entering")
...     raise RuntimeError("Teleport!")
...   except RuntimeError:
...     print("arrived")
...   print("leaving")
... 
>>> foo()
entering
arrived
leaving
```